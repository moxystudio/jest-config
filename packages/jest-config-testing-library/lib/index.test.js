'use strict';

const { withRTL, withNTL } = require('.');

describe('withRTL', () => {
    const buildConfig = () => ({
        testEnvironment: 'jsdom',
        setupFilesAfterEnv: ['foo'],
    });

    it('should match snapshot', () => {
        expect(withRTL()(buildConfig())).toMatchSnapshot();
    });

    it('should throw if testEnvironment is not jsdom', () => {
        const config = {
            ...buildConfig(),
            testEnvironment: 'node',
        };

        expect(() => withRTL()(config)).toThrow(/expected testEnvironment to be set to "jsdom"/i);
    });
});

describe('withNTL', () => {
    const buildConfig = () => ({
        testEnvironment: 'node',
        setupFilesAfterEnv: ['foo'],
        setupFiles: ['bar'],
        snapshotSerializers: ['baz'],
    });

    it('should match snapshot', () => {
        expect(withNTL()(buildConfig())).toMatchSnapshot();
    });

    it('should throw if testEnvironment is not node', () => {
        const config = {
            ...buildConfig(),
            testEnvironment: 'jsdom',
        };

        expect(() => withNTL()(config)).toThrow(/expected testEnvironment to be set to "node"/i);
    });
});

