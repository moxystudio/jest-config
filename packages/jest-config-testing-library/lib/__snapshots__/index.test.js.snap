// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withNTL should match snapshot 1`] = `
Object {
  "setupFilesAfterEnv": Array [
    "foo",
    "<PROJECT_ROOT>/node_modules/@testing-library/jest-native/extend-expect.js",
  ],
  "testEnvironment": "node",
}
`;

exports[`withRTL should match snapshot 1`] = `
Object {
  "setupFilesAfterEnv": Array [
    "foo",
    "<PROJECT_ROOT>/node_modules/@testing-library/jest-dom/extend-expect.js",
  ],
  "testEnvironment": "jsdom",
}
`;
